perf/perfevents_24_7.py

```diff
@@ -0,0 +1,191 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Athira Rajeev<atrajeev@linux.vnet.ibm.com>
+
+import os
+import sys
+import re
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+
+
+class test_eliminate_domain_suffix(Test):
+    """
+    This tests domain name suffix in event names
+    """
+
+    def setUp(self):
+        """
+        Check if kernel version has the supported feature
+        This feature is enabled from 3.10.0.419
+        """
+
+        # Compare kernel version
+        kernel_ver = os.uname()[2]
+        split_value = kernel_ver.split('.')
+        kernel_Version = ".".join(split_value[0:3]).replace("-", ".")
+        kernel_to_compare = "3.10.0.419"
+
+        result_comparison = 0
+
+        for cur_ker, supported_ker in zip(kernel_Version.split(".")[:4],
+                                          kernel_to_compare.split(".")[:4]):
+            if cur_ker > supported_ker:
+                result_comparison = 1
+
+        if result_comparison != 1:
+            raise self.log.warn(
+                "This test is not supported on this kernel version")
+
+    def test(self):
+        nfail = 0
+
+        # Check if this is a guest
+        # 24x7 is not suported on guest
+        cpu_output = process.run("cat /proc/cpuinfo")
+        if "emulated by" in cpu_output.stdout:
+            self.skipTest("This test is not supported on guest")
+
+        # Check if 24x7 is present
+        if os.path.exists("/sys/bus/event_source/devices/hv_24x7"):
+            self.log.info('/sys/bus/event_source/devices/hv_24x7 present')
+        else:
+            raise self.error("hv_24x7 doesn't exist.
+                             This feature is supported on only lpar")
+
+        # Performance measurement has to be enabled in lpar through BMC
+        # Check if its enabled
+        # Refer https://bugzilla.linux.ibm.com/show_bug.cgi?id=139404#c21
+        result_perf = process.run("perf stat -v -C 0 -e\
+            hv_24x7/HPM_0THRD_NON_IDLE_CCYC,domain = 2,\
+            core = 1/ sleep 1", ignore_status=True)
+        if "You may not have permission to collect stats" in\
+                result_perf.stderr:
+            raise self.error("Please enable lpar to allow\
+                             collecting the 24x7 counters info")
+
+        # Testing feature: Display domain indices in sysfs
+        # Part of commit d34171e
+        EVENT_SYSFS = "/sys/bus/event_source/devices/hv_24x7/events"
+        pattern = re.compile('1: Physical Chip\n2: Physical Core\n
+                             3: VCPU Home Core\n4: VCPU Home Chip\n
+                             5: VCPU Home Node\n6: VCPU Remote Node')
+        result = process.run(
+            'cat /sys/bus/event_source/devices/hv_24x7/interface/domains')
+        Result_search = pattern.search(result.stdout)
+        if Result_search:
```

Comment:
It is preferred that variable names should be of lower case, with underscores separating words